src/main/java/com/rackspacecloud/metrics/tenantroutingservice/services/RoutingServiceImpl.java

```diff
@@ -48,6 +49,11 @@ public TenantRoutes getIngestionRoutingInformation(String tenantId, String measu
 
         Optional<TenantRoutes> routingInfo = routingInformationRepository.findById(tenantIdAndMeasurement);
 
+        if(!routingInfo.isPresent() && readOnly) {
```

Comment:
You can move this body of code into the below `if` block. First expression is same for both of the `if` block. You can structure this code block as below:
```
if(!routingInfo.isPresent()) {
    if(readOnly) {
        ...
    }
```